src/seedu/addressbook/logic/Logic.java

```diff
@@ -85,10 +86,15 @@ public CommandResult execute(String userCommandText) throws Exception {
     private CommandResult execute(Command command) throws Exception {
         command.setData(addressBook, lastShownList);
         CommandResult result = command.execute();
-        storage.save(addressBook);
+        saveToStorageIfCommandIsMutating(command, addressBook);
         return result;
     }
 
+    private void saveToStorageIfCommandIsMutating(Command command, AddressBook addressBook) throws StorageFile.StorageOperationException {
+        if (command.isMutating())
```

Comment:
coding standard violation here; check how to write conditionals with a single line of code.